wallet/src/multisig_account.rs

```diff
@@ -0,0 +1,232 @@
+            .sum();
+
+        let mut public_key_bytes: [u8; PublicKey::SIZE] = [0u8; PublicKey::SIZE];
+        public_key_bytes.copy_from_slice(delinearized_pk_sum.compress().as_bytes());
+        PublicKey::from(public_key_bytes)
+    }
+
+    /// Creates a partial signature of the provided transaction.
+    pub fn partially_sign_transaction(
+        &self,
+        transaction: &Transaction,
+        public_keys: &[PublicKey],
+        commitments: &[Commitment],
+        secret: &RandomSecret,
+    ) -> PartialSignature {
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
+
+        self.key_pair
+            .partial_sign(
+                &sorted_public_keys,
+                secret,
+                commitments,
+                transaction.serialize_content().as_slice(),
+            )
+            .0
+    }
+
+    /// Creates a signature proof. Panics if the length of `signatures` is not equal to `min_signatures`.
+    pub fn create_proof(
+        &self,
+        aggregated_public_key: &PublicKey,
+        aggregated_commitment: &Commitment,
+        signatures: &[PartialSignature],
+    ) -> SignatureProof {
+        assert!(
+            signatures.len() == self.min_signatures as usize,
+            "Number of signatures must be the same as the minimal signatures."
+        );
+
+        let aggregated_signature: PartialSignature = signatures.iter().sum();
+        let raw_aggregated_signature = aggregated_signature.as_bytes();
+        let raw_aggregated_commitment = aggregated_commitment.to_bytes();
+
+        let mut combined =
+            Vec::with_capacity(raw_aggregated_signature.len() + raw_aggregated_commitment.len());
+        combined.extend_from_slice(&aggregated_commitment.to_bytes());
+        combined.extend_from_slice(aggregated_signature.as_bytes());
+        let signature = Signature::from_bytes(&combined).expect("Invalid signature constructued.");
+
+        SignatureProof {
+            merkle_path: MerklePath::new::<Blake2bHasher, PublicKey>(
+                &self.public_keys,
+                aggregated_public_key,
+            ),
+            public_key: *aggregated_public_key,
+            signature,
+        }
+    }
+
+    /// Signs the transaction.
+    pub fn sign_transaction(
+        &self,
+        transaction: &Transaction,
+        aggregated_public_key: &PublicKey,
+        aggregated_commitment: &Commitment,
+        partial_signatures: &[PartialSignature],
+    ) -> Transaction {
+        let proof = self.create_proof(
+            aggregated_public_key,
+            aggregated_commitment,
+            partial_signatures,
+        );
+
+        let mut signed_transaction = transaction.clone();
+        signed_transaction.proof = proof.serialize_to_vec();
+
+        signed_transaction
+    }
+
+    fn combinations<T: Clone>(list: &[T], k: usize) -> Vec<Vec<T>> {
```

Comment:
Can't we use [itertool's combinations](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.combinations)?